plugins/image-prioritizer/class-image-prioritizer-video-tag-visitor.php

```diff
@@ -123,4 +128,48 @@ private function preload_poster_image( OD_Tag_Visitor_Context $context ): bool {
 
 		return true;
 	}
+
+	/**
+	 * Adjust `autoplay` and `preload` values for videos outside initial viewport.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param OD_Tag_Visitor_Context $context Tag visitor context, with the cursor currently at an embed block.
+	 * @return bool Whether the tag should be tracked in URL metrics.
+	 */
+	private function lazy_load_videos( OD_Tag_Visitor_Context $context ): bool {
+		$processor = $context->processor;
+
+		$xpath = $processor->get_xpath();
+
+		$intersection_ratio = $context->url_metric_group_collection->get_element_max_intersection_ratio( $xpath );
+
+		if ( $intersection_ratio > 0 ) {
+			return false;
+		}
+
+		$preload = $processor->get_attribute( 'preload' );
+		if ( 'none' !== $preload ) {
+			$processor->set_attribute( 'data-original-preload', null !== $preload ? $preload : 'default' );
+			$processor->set_attribute( 'preload', 'none' );
+		}
+
+		if ( true === $processor->get_attribute( 'autoplay' ) ) {
```

Comment:
In HTML `autoplay="true"` also will enable autoplay, same as `autoplay="false"`, since this is a boolean attribute. For best compatibility, this should rather be:

```suggestion
		if ( null !== $processor->get_attribute( 'autoplay' ) ) {
```